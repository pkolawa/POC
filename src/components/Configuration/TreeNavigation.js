import React from "react";
import T from 'prop-types';
import { withStyles } from "@material-ui/core/styles";
import TreeView from "@material-ui/lab/TreeView";
import TreeItem from "@material-ui/lab/TreeItem";
import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
import ChevronRightIcon from "@material-ui/icons/ChevronRight";

const styles = (theme) => ({
    
});

class TreeNavigation extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            formFields: []
        }
    }

    static propTypes = {
        treeItems: T.array.isRequired,
        nodeClickHandler: T.func.isRequired
	};

    getTreeItemsFromData = treeItems => {
        return treeItems.map(treeItemData => {
            let children = undefined;
            if (treeItemData.children && treeItemData.children.length > 0) {
                children = this.getTreeItemsFromData(treeItemData.children);
            }

            return (
                <TreeItem
                    key={treeItemData.id}
                    nodeId={treeItemData.id}
                    label={this.renderlabel(treeItemData)}
                    children={children}
                />
            );
        });
    };

    renderlabel = item => {
        const { name, id, data } = item;

        return (
            <span onClick={() => this.showDetails(data)} nodeId={id} key={id}>{name}</span>
        )
    }

    showDetails = (data) => {
        const { nodeClickHandler } = this.props;
        nodeClickHandler(data);

    }

    dataTreeView = (treeItems) => {
        return (
            <TreeView
                defaultCollapseIcon={<ExpandMoreIcon />}
                defaultExpandIcon={<ChevronRightIcon />}
            >
                {this.getTreeItemsFromData(treeItems)}
            </TreeView>
        );
    };

    render() {
        const { treeItems } = this.props;
        
        return this.dataTreeView(treeItems);

    }
}

export default withStyles(styles)(TreeNavigation);
