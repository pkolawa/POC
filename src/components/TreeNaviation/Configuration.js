import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";
import TreeNavigation from "./TreeNavigation";
import metaData from "./meta-data-config.json";
import FormComponent from "./FormComponent";
import Menu from "../Menu";

const styles = (theme) => ({
  root: {
    display: "flex",
    flexWrap: "wrap",
    flexDirection: "row",
  },
  container: {
    flexBasis: `calc(100% - 240px)`,
  },
  gridContainer: {
      marginTop: theme.spacing(3)
  }
});

class Configuration extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      formData: [],
      // metaData: []
    };
  }

  componentDidMount() {
    //fetch('http://localhost:3001/configuration').then( metaData => this.setState({metaData})).catch(e => console.error(e))
  }

  handleNodeClick = (formData) => {
    if (formData && formData.length > 0) {
      this.setState({ formData });
    }
  };

  render() {
    const { classes } = this.props;
    const { formData } = this.state;

    return (
      <div className={classes.root}>
        <Menu />
        <Container className={classes.container}>
          <Typography align="center" variant="h2" color="secondary">
            Configuration Page
          </Typography>
          <Grid container className={classes.gridContainer} spacing={3}>
            <Grid item xs={3}>
              <TreeNavigation
                treeItems={metaData}
                nodeClickHandler={this.handleNodeClick}
              />
            </Grid>
            <Grid item xs={9}>
              <FormComponent pageData={formData} />
            </Grid>
          </Grid>
        </Container>
      </div>
    );
  }
}

export default withStyles(styles)(Configuration);
