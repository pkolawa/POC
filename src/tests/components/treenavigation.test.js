import React from "react";
import { shallow } from "enzyme";
import { TreeNavigation } from "../../components/TreeNaviation/TreeNavigation";

const metaData = [
  {
    id: "1",
    name: "serverpools",
    children: [
      {
        id: "2",
        name: "pools",
        children: [
          {
            id: "3",
            name: "LocalPool",
            data: [
              {
                fields: [
                  {
                    name: "id",
                    type: "string",
                    value: "local1",
                  },
                  {
                    name: "key",
                    type: "string",
                    value: "A",
                  },
                  {
                    name: "hostname",
                    type: "string",
                    value: "loalhost1",
                  },
                  {
                    name: "port",
                    type: "number",
                    value: "8091",
                  },
                ],
              },
              {
                fields: [
                  {
                    name: "id",
                    type: "string",
                    value: "local2",
                  },
                  {
                    name: "key",
                    type: "string",
                    value: "B",
                  },
                  {
                    name: "hostname",
                    type: "string",
                    value: "loalhost2",
                  },
                  {
                    name: "port",
                    type: "number",
                    value: "8092",
                  },
                ],
              },
            ],
          },
          {
            id: "4",
            name: "TestPool",
            data: [
              {
                fields: [
                  {
                    name: "id",
                    type: "string",
                    value: "test1",
                  },
                  {
                    name: "key",
                    type: "string",
                    value: "A",
                  },
                  {
                    name: "hostname",
                    type: "string",
                    value: "loalhost1",
                  },
                  {
                    name: "port",
                    type: "number",
                    value: "9010",
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
  },
  {
    id: "5",
    name: "redisconfig",
    data: [
      {
        fields: [
          {
            name: "enabled",
            type: "string",
            value: "false",
          },
          {
            name: "mode",
            type: "string",
            value: "single",
          },
          {
            name: "hosts",
            type: "string",
            value: "localhost:6379",
          },
          {
            name: "cluster_name",
            type: "string",
            value: "abcCluster",
          },
          {
            name: "database",
            type: "number",
            value: "0",
          },
        ],
      },
    ],
  },
];
let nodeClickHandler;
describe("Test TreeNavigation", () => {
  beforeAll(() => {
    nodeClickHandler = jest.fn();
  });
  test("Should Render TreeNavigation", () => {
    const wrapper = shallow(
      <TreeNavigation
        treeItems={metaData}
        nodeClickHandler={nodeClickHandler}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
