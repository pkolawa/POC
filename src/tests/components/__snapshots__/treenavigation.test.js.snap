// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TreeNavigation Should Render TreeNavigation 1`] = `
<WithStyles(ForwardRef(TreeView))
  defaultCollapseIcon={<UNDEFINED />}
  defaultExpandIcon={<UNDEFINED />}
>
  <WithStyles(ForwardRef(TreeItem))
    key="1"
    label={
      <span
        nodeId="1"
        onClick={[Function]}
      >
        serverpools
      </span>
    }
    nodeId="1"
  >
    <WithStyles(ForwardRef(TreeItem))
      key="2"
      label={
        <span
          nodeId="2"
          onClick={[Function]}
        >
          pools
        </span>
      }
      nodeId="2"
    >
      <WithStyles(ForwardRef(TreeItem))
        key="3"
        label={
          <span
            nodeId="3"
            onClick={[Function]}
          >
            LocalPool
          </span>
        }
        nodeId="3"
      />
      <WithStyles(ForwardRef(TreeItem))
        key="4"
        label={
          <span
            nodeId="4"
            onClick={[Function]}
          >
            TestPool
          </span>
        }
        nodeId="4"
      />
    </WithStyles(ForwardRef(TreeItem))>
  </WithStyles(ForwardRef(TreeItem))>
  <WithStyles(ForwardRef(TreeItem))
    key="5"
    label={
      <span
        nodeId="5"
        onClick={[Function]}
      >
        redisconfig
      </span>
    }
    nodeId="5"
  />
</WithStyles(ForwardRef(TreeView))>
`;
